processing/src/main/java/io/druid/query/groupby/GroupByQueryHelper.java

```diff
@@ -64,12 +66,16 @@
   )
   {
     final GroupByQueryConfig querySpecificConfig = config.withOverrides(query);
-    final QueryGranularity gran = query.getGranularity();
+    final Granularity gran = query.getGranularity();
     final long timeStart = query.getIntervals().get(0).getStartMillis();
 
     // use gran.iterable instead of gran.truncate so that
     // AllGranularity returns timeStart instead of Long.MIN_VALUE
-    final long granTimeStart = gran.iterable(timeStart, timeStart + 1).iterator().next();
+    long granTimeStart = timeStart;
+    if (!(gran instanceof AllGranularity)) {
```

Comment:
Does this need to be `instanceof`?  In other parts of the code, we do `Granularity.ALL.equals()`, does that work?